owncloudApp/src/main/java/com/owncloud/android/extensions/ViewModelExt.kt

```diff
@@ -37,6 +37,26 @@ object ViewModelExt : KoinComponent {
 
     private val contextProvider: ContextProvider by inject()
 
+    fun <T, Params> ViewModel.runAsyncUseCase(
+        coroutineDispatcher: CoroutineDispatcher,
+        requiresConnection: Boolean = true,
+        useCase: BaseUseCaseWithResult<T, Params>,
+        useCaseParams: Params
+    ) {
+        viewModelScope.launch(coroutineDispatcher) {
+
+            // If usecase requires connection and is not connected, it is not needed to execute use case.
+            if (requiresConnection and !contextProvider.isConnected()) {
+                Timber.w("${useCase.javaClass.simpleName} will not be executed due to lack of network connection")
+                return@launch
+            }
+
+            val useCaseResult = useCase.execute(useCaseParams)
```

Comment:
Don't you do anything with this `useCaseResult`? How the user is notified about the usecase not being executed because of lack of connection?

Reply:
Seems like i forgot to remove this function. Finally i used the `runUseCaseWithResult` one. Thanks